src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -383,24 +383,28 @@ def _await_fluent_launch(
 
 
 def _get_server_info(
-    server_info_file_name: str,
+    server_info_file_name: Optional[str] = None,
     ip: Optional[str] = None,
     port: Optional[int] = None,
     password: Optional[str] = None,
 ):
     """Get server connection information of an already running session."""
-    if ip and port:
+    if not (ip and port) and not server_info_file_name:
+        raise RuntimeError(
+            "Please provide either ip and port data or server-info file."
+        )
+    if ip and port and server_info_file_name:
         logger.warning(
             "Could not parse server-info file because ip and port were provided explicitly."
```

Comment:
For all of the following combinations:
```
(ip=..., port=..., server_info_file_name=...)
(ip=..., server_info_file_name=...)
(port=..., server_info_file_name=...)
```
we take the ip and port values from server-info file. This is the correct behaviour for Fluent connection as we cannot use a different ip/port on the client side when a server-info file is available. When the client tries to connect to Fluent, Fluent will match the ip/port values sent from the client against what it has written in the server-info file.

So, the above message can be changed to something like: `The ip and port will be extracted from the server-info file and their explicitly specified values will be ignored.`

Reply:
@mkundu1 https://github.com/ansys/pyfluent/pull/2176/files#diff-e49ebe5caff347c437627b296d498b5c53873625992b95779b6bacb1965e5f07.